tests/testthat/test-FilteredData.R

```diff
@@ -345,35 +340,49 @@ testthat::test_that(
   }
 )
 
-datasets <- FilteredData$new()
-adsl <- as.data.frame(as.list(setNames(nm = c(teal.data::get_cdisc_keys("ADSL")))))
-adsl$sex <- c("F")
-datasets$set_dataset(teal.data::cdisc_dataset("ADSL", adsl))
-datasets$set_dataset(teal.data::dataset("mock_iris", head(iris)))
-utils::data(miniACC, package = "MultiAssayExperiment")
-datasets$set_dataset(teal.data::dataset("miniACC", miniACC))
+
+get_filtered_data_object <- function() {
+  utils::data(miniACC, package = "MultiAssayExperiment")
+  adsl <- as.data.frame(as.list(setNames(nm = c(teal.data::get_cdisc_keys("ADSL")))))
+  adsl$sex <- c("F")
+
+  CDISCFilteredData$new(
+    list(
+      ADSL = list(dataset = adsl, keys = teal.data::get_cdisc_keys("ADSL"), parent = character(0)),
+      mock_iris = list(dataset = head(iris)),
+      miniACC = list(dataset = miniACC)
+    ),
+    keys = NULL
+  )
+}
+
 
 testthat::test_that("get_filter_overview accepts all datasets argument input", {
+  datasets <- get_filtered_data_object()
   testthat::expect_error(isolate(datasets$get_filter_overview("all")), NA)
 })
 
 testthat::test_that("get_filter_overview accepts single dataset argument input", {
+  datasets <- get_filtered_data_object()
   testthat::expect_error(isolate(datasets$get_filter_overview("ADSL")), NA)
   testthat::expect_error(isolate(datasets$get_filter_overview("mock_iris")), NA)
   testthat::expect_error(isolate(datasets$get_filter_overview("miniACC")), NA)
 })
 
 testthat::test_that("get_filter_overview throws error with empty argument input", {
+  datasets <- get_filtered_data_object()
   testthat::expect_error(isolate(datasets$get_filter_overview()), "argument \"datanames\" is missing, with no default")
 })
 
 testthat::test_that("get_filter_overview throws error with wrong argument input", {
+  datasets <- get_filtered_data_object()
   testthat::expect_error(isolate(datasets$get_filter_overview("AA")), "Some datasets are not available:")
   testthat::expect_error(isolate(datasets$get_filter_overview("")), "Some datasets are not available:")
   testthat::expect_error(isolate(datasets$get_filter_overview(23)), "Some datasets are not available:")
 })
 
 testthat::test_that("get_filter_overview returns overview matrix for non-filtered datasets", {
+  datasets <- get_filtered_data_object()
```

Comment:
So this test and the following tests fail.

This is what is expected:

![image](https://user-images.githubusercontent.com/15201933/174114296-d168c830-f0e5-47b2-b1d7-e7f4ea0804f5.png)

But we can no longer distinguish that `mock_iris` is not CDISC and `ADSL` is CDISC with no parent. Previously the former was `DefaultFilteredDataset` and the latter `CDISCFilteredDataset`. Now they are both `DefaultFilteredDataset` and then either they both have subjs as this branch now creates a CDISCFilteredData object:
 
![image](https://user-images.githubusercontent.com/15201933/174114998-9e9732df-a19b-4988-b308-a4ffc2a39009.png)

Or if I change the creation of datasets to use FilteredData we get neither with subjs:

![image](https://user-images.githubusercontent.com/15201933/174115313-71c5fa36-c97e-4b0c-9e90-2c593dcadc31.png)

Is this change of behaviour OK? More generally it seems like the design we've changed to is fighting against the dispatch mechanisms in R (we have cdisc, non cdisc and MAE data, on the dataset level we have MAE and not MAE and on the data level we have CDISC and not CDISC). I'm OK with this if it's a stepping stone to simplifying the filter panel and removing all the data but I don't really like it  Especially trying to understand the logic to get the filter information - maybe this logic can be simplified?